src/prefect/tasks/transform/transform_tasks.py

```diff
@@ -0,0 +1,207 @@
+        self.mql_server_url = mql_server_url
+        self.mql_server_url_env_var = mql_server_url_env_var
+        self.model_key_id = model_key_id
+        self.materialization_name = materialization_name
+        self.start_time = start_time
+        self.end_time = end_time
+        self.output_table = output_table
+        self.force = force
+        self.use_async = use_async
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "api_key",
+        "api_key_env_var",
+        "mql_server_url",
+        "mql_server_url_env_var",
+        "model_key_id",
+        "materialization_name",
+        "start_time",
+        "end_time",
+        "output_table",
+        "force",
+        "use_async",
+    )
+    def run(
+        self,
+        api_key: str = None,
+        api_key_env_var: str = None,
+        mql_server_url: str = None,
+        mql_server_url_env_var: str = None,
+        model_key_id: int = None,
+        materialization_name: str = None,
+        start_time: str = None,
+        end_time: str = None,
+        output_table: str = None,
+        force: bool = False,
+        use_async: bool = False,
+    ):
+        """
+        Task run method to create a materialization against a Transform metrics
+        layer deployment.
+        All parameters can be provided either during task initialization or directly
+        in this `run` method.
+
+        Args:
+            - api_key (str, optional): Transform API Key to be used to
+                connect to Transform MQL Server.
+            - api_key_env_var (str, optional): The name of the environment variable
+                that contains the API Key to be used to connect to Transform MQL Server.
+            - mql_server_url (str, optional): The URL of the Transform MQL Server
+                from which to create the materialization.
+            - mql_server_url_env_var (str, optional): The name of the environment variable
+                that contains the URL of the Transform MQL Server from which to
+                create the materialization.
+            - model_key_id (int, optional): The unique identifier of the Transform model
+                against which the transformation will be created.
+            - materialization_name (str, optional): The name of the Transform materialization
+                to create.
+            - start_time (str, optional): The UTC start time of the materialization.
+            - end_time (str, optional): The ISO end time of the materialization.
+            - output_table (str, optional): The name of the database table, in the form of
+                `schema_name.table_name`, where the materialization will be created.
+            - force (bool, optional): Whether to force the materialization creation
+                or not. Defaults to `False`.
+            - use_async (bool, optional): Whether to trigger the materialization creation
+                in async mode or not. Defaults to `False`.
```

Comment:
In Transform, there are two ways of creating/refreshing a materialization using the `MQLClient`:

- by using `materialize` --> this will call the Transform server and the client will wait for a response from the Transform server. This is the method that the `TransformCreateMaterialization` task calls under the hood in this task when invoked with `use_async = False`.
- by using `create_materialization` --> this will call the Transform server which will queue the request and will return back a response containing the status of the queued operation. This is the method that the `TransformCreateMaterialization` task calls under the hood in this task when invoked with `use_async = True`.

Maybe @andykram can jump in and provide some additional details 😊